app/features-json/setup_json_controller.rb

```diff
@@ -40,5 +40,33 @@ class SetupJSONController < APIController
         }
       )
     end
+
+    post "#{HOME}/oauth", authenticate: false do
+      client_id = params["client_id"]
+      client_secret = params["client_secret"]
+
+      if client_id.nil? || client_secret.nil?
+        json_error!(
+          error_message: "Must provide both the OAuth client ID and client secret",
```

Comment:
Ohh, sooo to avoid having to create a personal access token, we now ask people to create their own GitHub OAuth app? And we can't create one for everyone, as the callback URL won't work?

Reply:
>we now ask people to create their own GitHub OAuth app? 

yea!

>we can't create one for everyone, as the callback URL won't work?

the callback url won't work and the user won't own the oauth app.